### ISSUE TITLE
Updating registry to use dev location doesn't seem to work

### ISSUE THREAD

**Issue by alevinetx** [stevencohn/OneMore#566]
**How can I help you?**
I am trying to use your registry update script to improve the dev/test cycle, but it does not seem to want to work for me

1: Build solution
2: Run debug installer
3: Run registry update script
4: Open OneNote -- plugin is there
5:  Exit OneNote, rename(delete) debug DLL
6: Open OneNote -- plugin is there
7: Exit OneNote, rename(delete) DLL from  \program files\....
8 Open OneNote -- plugin is missing 

FWIW, I did find 2 additional references to the installed DLL.  I added them to the update script, but still nothing.
- Registry::HKEY_CLASSES_ROOT\WOW6432Node\CLSID\$guid\InprocServer32
- Registry::HKEY_CLASSES_ROOT\WOW6432Node\CLSID\$guid\InprocServer32\$pv

Do you have any additional suggestions for getting this to work?

Thank you!

VS 2022
Edition	Windows 10 Enterprise
Version	21H2
Installed on	‎8/‎19/‎2021
OS build	19044.1766
Experience	Windows Feature Experience Pack 120.2212.4180.0

**Comment by stevencohn** [comment]
Silly questions but I have to ask
1. Are you running it from an elevated PowerShell prompt?
2. Are you running it from (current working directory) the OneMore solution folder?
The two additional references you found are already in the script [here](https://github.com/stevencohn/OneMore/blob/02eef127c76d516e97a89fe1286e526aad0b3511/setdevreg.ps1#L46) and [here](https://github.com/stevencohn/OneMore/blob/02eef127c76d516e97a89fe1286e526aad0b3511/setdevreg.ps1#L55)

So do you have the latest version?

**Comment by alevinetx** [comment]
Please ask away
1:  Yes, as I'd tried it once w/o elevated w/o realizing, and it failed
2:  Yes. I've checked the registry entries and they're correct

I'm not building from master, but from 4.18.   Let me try again with the script versions you mentioned.

Other than that, I assume I have the process down.

**Comment by stevencohn** [comment]
You should only to run the installer once just to set up everything.

Build once.
Run the script.
Then you should be able to build again and again

If you monitor the OneMore log file, you can press Shift-F8 at any time to dump diagnostics to the log file. The Addin path should be the second line so you can confirm the location of the plugin you're running.

**Comment by alevinetx** [comment]
I looked at the script version, and it's the same
The 2 additional nodes I found are  the ...\WOW6432Node\ entries

**Comment by stevencohn** [comment]
64-bit Windows?
Have you run the installer with the same bit architecture as Windows?
The default build for the Setup project is set to x64. You can use build.ps1 to build either or both architecture installers.
Is your VS solution config set to Debug/AnyCPU?

**Comment by alevinetx** [comment]
1:  yes, 64b windows
2: 
![image](https://user-images.githubusercontent.com/5067356/186041887-f21b192d-9d3b-406b-8ad9-6240bafb266a.png)
![image](https://user-images.githubusercontent.com/5067356/186041971-8d8be482-e82a-4c49-81fb-6dc29fc84904.png)

**Comment by alevinetx** [comment]
I've managed to muck something up nicely.  Running the install, like I did prior, is working but the plugin fails to start on launch.
Scrubbing my registry to make sure I didn't leave some gunk

**Comment by alevinetx** [comment]
okay, a reboot and we're up again
1:Installed from setup/debug/ .msi
2: opened OneNote, shift-f8, verified it was in \program files\
3: close all OneNote instances
4: Ran the registry script, elevated
5: Restarted, dump, and it's still pointing to \program files\

```
22488:1] Log restarted
22488:7] Running command DiagnosticsCommand
Diagnostics.Execute()
--------------------------------------------------------------------------------
ONENOTE...: C:\Program Files (x86)\Microsoft Office\root\Office16\ONENOTE.EXE (16.0.14931.20604)
Addin path: C:\Program Files\River\OneMoreAddIn\River.OneMoreAddIn.dll
```

**Comment by alevinetx** [comment]
What if I provided the working debug directory to the installer?  Any issues you can think of ?

**Comment by stevencohn** [comment]
Probably the simplest solution is to install it, get it working, then just manually change the path to these two assemblies in the Registry:

* River.OneMoreAddin.dll 
* OneMoreProtocolHandler.exe

That's really all the script does

**Comment by alevinetx** [comment]
That's the route I took. I lost count, but total replacements between the 2 was just under 10, I believe.  
That difference may be why the update script wasn't proving successful for me.

But, for now, I can iterate and see changes with a restart. That's a win. 

And thank you!

**Comment by alevinetx** [comment]
side note, I can open this as an issue.. minor typo
OneMore/Properties/Resources.resx

line315, 'search'
```
  <data name="ribSearchMenu_Label" xml:space="preserve">
    <value>Seach and Tag</value>
    <comment>ribbon menu item</comment>
  </data>
```

**Comment by stevencohn** [comment]
> That's the route I took. I lost count, but total replacements between the 2 was just under 10, I believe. That difference may be why the update script wasn't proving successful for me.
> 
> But, for now, I can iterate and see changes with a restart. That's a win.
> 
> And thank you!

My guess is that you started with the wrong installer. As per the [installation instructions](https://github.com/stevencohn/OneMore#install), the installer - x86 or x64 - is tied to Windows and has nothing to do with the version of Office or OneNote.

**Comment by stevencohn** [comment]
Note there is an iq.ps1 script that validate the configuration. It will (should) tell you what's misaligned

**Comment by alevinetx** [comment]
> My guess is that you started with the wrong installer. As per the [installation instructions]


I've used the same installer from day 0, the same one I used again last night.   

I'll check out that script, tho!

**Comment by stevencohn** [comment]
I've updated the iq.ps1 script with verbose output

PS> `.\setdevreg.ps1`
PS> `.\iq.ps1 -verbose`

This should tell you all you need to know.

**Comment by stevencohn** [comment]
![image](https://user-images.githubusercontent.com/9964757/187685176-c9e4d272-7d1b-4755-a093-91a94527af97.png)

**Comment by alevinetx** [comment]
I'm hesitant to do this with my current project, as it's working.  But I'll definitely be looking at this for the next one.

thank you for your continued support!
